metrics/ci_integrations/docs/features/in_progress_builds/in_progress_builds_introduction.md

```diff
@@ -0,0 +1,271 @@
+
+Also, the `GithubActionsSourceClientAdapter` should now implement the new `fetchBuild` method to match the `SourceClient` interface. The adapter should use the specified `GithubActionsClient` instance to fetch the build with the given number.
+
+#### Destination Client
+
+The `DestinationClient` is an interface for `destination` clients that provide abilities to store synchronized builds and interact with the database. As mentioned in the [Sync Algorithm](#sync-algorithm) section, the in-progress builds introduction requires adding the re-syncing in-progress builds stage into the sync algorithm. This stage queries in-progress builds from the database and then updates them after re-syncing. Thus, the `DestinationClient` should be updated with appropriate methods to perform the required functionality.
+
+The following methods are to be added to the `DestinationClient` interface:
+
+- `fetchBuildsWithStatus` - fetches all builds having the given status for the given project.
+- `updateBuilds` - updates all the given builds with new values for the given project.
+
+The following class diagram demonstrates the structure of `destination` clients:
+
+![Destination clients class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/ci_integrations_in_progress_doc/metrics/ci_integrations/docs/features/in_progress_builds/diagrams/destination_clients_class_diagram.puml)
+
+The following subsection covers the details of changes to the Firestore integration.
+
+##### Firestore Integration
+
+The Firestore destination integration allows interacting with the Firestore database to manipulate synchronized builds data. The in-progress builds introduction requires the Firestore integration to be able to fetch in-progress builds for re-syncing and updating these builds after the re-syncing completes. The `DestinationClient` interface contains new methods required for the sync algorithm.
+
+The following methods are to be implemented for the `FirestoreDestinationClientAdapter` to match the `DestinationClient` interface:
+
+- `fetchBuildsWithStatus`
+- `updateBuilds`
+
+The `fetchBuildsWithStatus` is used by the sync algorithm to fetch in-progress builds stored in the database. This method should throw if the project with the given identifier does not exist. If the project exists but doesn't have synchronized builds yet, the method should return an empty list.
+
+The `updateBuilds` method is used to update in-progress builds with new data and status. This method should throw if the project with the given identifier does not exist. If updating one of the builds fails, the method should continue updating other builds, as the failed build can be re-synced later, during the next synchronization. Generally speaking, the implementation should consider the following cases:
```

Comment:
It would be great to clarify what does the `implementation should consider` actually mean?